src/mesh/boundary_info.C

```diff
@@ -1865,9 +1877,154 @@ BoundaryInfo::build_node_list_from_side_list()
 
           // Add each node node on the side with the side's boundary id
           for (unsigned int i=0; i<side->n_nodes(); i++)
-            this->add_node(side->node_ptr(i), pos->second.second);
+            {
+              const boundary_id_type bcid = pos->second.second;
+              this->add_node(side->node_ptr(i), bcid);
+              if (!mesh_is_serial)
+                {
+                  const processor_id_type proc_id =
+                    side->node_ptr(i)->processor_id();
+                  if (proc_id != my_proc_id)
+                    nodes_to_push[proc_id].insert
+                      (std::make_pair(side->node_id(i), bcid));
+                }
+            }
+        }
+    }
+
+  // If we're on a serial mesh then we're done.
+  if (mesh_is_serial)
+    return;
+
+  // Otherwise we need to push ghost node bcids to their owners, then
+  // pull ghost node bcids from their owners.
+  Parallel::MessageTag
+    node_pushes_tag = this->comm().get_unique_tag(31337),
```

Comment:
We'll see.  I'm using "867", "5309" and "42", "6*9" in an upcoming MOOSE PR, so I'm already nearing the bottom of the cutesy jokes barrel.